MonstieWash/Assets/Scripts/Eraser.cs

```diff
@@ -0,0 +1,233 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Eraser : MonoBehaviour
+{
+    [SerializeField] private Tool tool;
+
+    private List<Erasable> m_erasables = new();
+    private Vector2Int m_mouseScreenPos = Vector2Int.zero;
+    private Vector2Int m_prevMouseScreenPos = Vector2Int.zero;
+
+    private struct Erasable
+    {
+        public Erasable(GameObject obj, Sprite sprite, byte[] maskPixels)
+        {
+            this.obj = obj;
+            this.sprite = sprite;
+            this.maskPixels = maskPixels;
+        }
+
+        public GameObject obj;
+        public Sprite sprite;
+        public byte[] maskPixels;
+
+        public void ApplyMask()
+        {
+            var colors = sprite.texture.GetPixels();
+            var newColors = new Color[maskPixels.Length];
+
+            for (int i = 0; i < newColors.Length; i++)
+            {
+                newColors[i].r = colors[i].r;
+                newColors[i].g = colors[i].g;
+                newColors[i].b = colors[i].b;
+                newColors[i].a = Mathf.Min((255 - maskPixels[i])/255f, colors[i].a);
+            }
+
+            sprite.texture.SetPixels(newColors, 0);
+            sprite.texture.Apply(false);
+        }
+    }
+
+    private void Awake()
+    {
+        InitializeTool();
+        PopulateErasables();
+    }
+
+    private void Update()
+    {
+        transform.localScale = Vector3.one * tool.size;
+
+        if (!Input.GetMouseButton(0)) return;
+        if (!MouseMoved()) return;
+
+        foreach (var erasable in m_erasables)
+        {
+            if (UpdateErasableMask(erasable)) erasable.ApplyMask();
+        }
+    }
+
+    /// <summary>
+    /// Sets up the tool and related variables ready for use.
+    /// </summary>
```

Comment:
If we'll need to document our functions with XML tags it would be nice to have it as part of the conventions in the dev wiki

Reply:
Ah yep, good idea. I think I'll add it there as it's v helpful when reading through the code.